### PR TITLE
Add two more tests to the "service - call" suite

### PR DESCRIPTION
**Pull request by basyskom-meerkoetter**

### CODE REVIEW

File: tests/server/check_services_call.c [hunk 1]
```diff
@@ -103,6 +103,30 @@ START_TEST(callMethodAndObjectExistsButMethodHasWrongNodeClass) {
...
+START_TEST(callMethodOnUnrelatedObject) {
+    UA_CallMethodRequest callMethodRequest;
+    UA_CallMethodRequest_init(&callMethodRequest);
+    callMethodRequest.methodId = UA_NODEID_NUMERIC(0, UA_NS0ID_SERVER_REQUESTSERVERSTATECHANGE);
+    callMethodRequest.objectId = UA_NODEID_NUMERIC(0, UA_NS0ID_OBJECTSFOLDER);  // not connected via hasComponent
+
+    UA_CallMethodResult result;
+    UA_CallMethodResult_init(&result);
+    result = UA_Server_call(server, &callMethodRequest);
+    ck_assert_int_eq(result.statusCode, UA_STATUSCODE_BADMETHODINVALID);
```
Comment (basyskom-meerkoetter) [inline tests/server/check_services_call.c]: @jpfr do you have an idea why this assertion fails?

the method is not connected to the object via a hasComponent reference.
Still the code in ua_service_call.c:145-159 validates and doesn't return with BADMETHODINVALID.

Comment (jpfr) [reply tests/server/check_services_call.c]: In that case, can you set a breakpoint on `found = true;` in line 155?
This will give the mis-classified node.

Comment (basyskom-meerkoetter) [reply tests/server/check_services_call.c]: Sure, will try.
But the assumption is (so far) that the unit-test is correct and the code is ua_services_call.c has a bug?

Comment (basyskom-meerkoetter) [reply tests/server/check_services_call.c]: My fault. The test-case was wrong

**Pull request closed without merging**